contracts/meta-oracles/lib/RSILibrary.sol

```diff
@@ -59,26 +63,34 @@ library RSILibrary{
         uint256 positiveDataSum = 0;
         uint256 negativeDataSum = 0;
 
+        require(
+            _dataArray.length > 1,
+            "Length of data array must be greater than 1"
+        );
+
         for (uint256 i = 1; i < _dataArray.length; i++) {
-            // If current day price is greater than previous day's
-            if (_dataArray[i - 1] > _dataArray[i]) {
-                positiveDataSum = positiveDataSum.add(_dataArray[i - 1]).sub(_dataArray[i]);
+            uint256 currentPrice = _dataArray[i - 1];
+            uint256 previousPrice = _dataArray[i];
+            if (currentPrice > previousPrice) {
+                positiveDataSum = positiveDataSum.add(currentPrice).sub(previousPrice);
             }
             else {
-                negativeDataSum = negativeDataSum.add(_dataArray[i]).sub(_dataArray[i - 1]);
+                negativeDataSum = negativeDataSum.add(previousPrice).sub(currentPrice);
             }
         }
 
-        // Get rid of divide by 0
-        if (negativeDataSum == 0 && positiveDataSum == 0) {
-            negativeDataSum = 1;
-        }
+        require(
+            negativeDataSum > 0 || positiveDataSum > 0,
+            "Not valid RSI Value"
```

Comment:
Our error messaging conventions are `contractName.functionName` so here `RSILibrary.calculate`. Fix for above require as well.